test/hotspot/jtreg/runtime/jni/daemonDestroy/TestDaemonDestroy.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @bug 8290732
+ * @comment Test uses custom launcher that attempts to destroy the VM on both
+ *          a daemon and non-daemon thread. The result should be the same in
+ *          both cases.
+ * @requires vm.flagless
+ * @library /test/lib
+ * @build Main
+ * @run main/native TestDaemonDestroy
+ * @run main/native TestDaemonDestroy daemon
+ */
+
+// Logic copied from SigTestDriver
+
+import jdk.test.lib.Platform;
+import jdk.test.lib.Utils;
+import jdk.test.lib.process.OutputAnalyzer;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+
+public class TestDaemonDestroy {
+
+    public static void main(String[] args) throws IOException {
+        Path launcher = Paths.get(Utils.TEST_NATIVE_PATH)
+            .resolve("daemonDestroy" + (Platform.isWindows() ? ".exe" : ""))
+            .toAbsolutePath();
+
+        System.out.println("Launcher = " + launcher +
+                           (Files.exists(launcher) ? " (exists)" : " (missing)"));
+
```

Comment:
Nit: better arg handling?
- If you want to just have one char arg, pass 'D' and 'N' for demon/nondemon mode instead of "1" and "".
- Instead of using args.length, maybe actually parse the input argument, or specify D and N there too and pass it on.

Reply:
I reworked it slightly so the Java side doesn't pass "" and the native code interprets any additional arg as "use daemon".